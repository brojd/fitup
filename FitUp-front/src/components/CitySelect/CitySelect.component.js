import React, { PropTypes } from 'react';
import classNames from 'classnames';
import styles from './CitySelect.stylesheet.css';
import uniqBy from 'lodash/uniqBy';
import compact from 'lodash/compact';

class CitySelect extends React.Component {
  constructor() {
    super();
    this._handleCityChange = this._handleCityChange.bind(this);
  }
  _getUniqCities() {
    let cities = this.props.gyms.map((gym) => {
      return gym.address.city;
    });
    cities.push(this.props.userLocation);
    return compact(uniqBy(cities));
  }
  _handleCityChange(e) {
    this.props.onChange(e.target.value);
  }
  render() {
    return (
      <select
        onChange={this._handleCityChange}
        className={classNames('uk-float-right', styles.citySelect)}
        defaultValue={this.props.userLocation} >
        {this._getUniqCities().map((city, i) => {
          return (
            <option className={styles.cityOption}
                    value={city}
                    key={i}>
              {city}
            </option>
          );
        })}
      </select>
    );
  }
}

CitySelect.propTypes = {
  onChange: PropTypes.func,
  gyms: PropTypes.array,
  userLocation: PropTypes.string
};

CitySelect.defaultProps = {
  gyms: []
};

export default CitySelect;
