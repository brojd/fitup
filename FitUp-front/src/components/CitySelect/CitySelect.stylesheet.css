.citySelect {
  height: 17px;
  border: none;
  border-bottom: 2px solid rgba(189, 0, 0, 0.2);
  color: #f9f9f9;
  background: linear-gradient(to bottom right, #400000, rgba(169, 0, 0, 0.85));
  border-radius: 3px;
  font-size: 11px;
  z-index: 5;
  width: 17%; }

.citySelect:hover {
  cursor: pointer; }

.citySelect:focus {
  outline: 0; }

.cityOption {
  color: #000000;
  background: #FFFFFF;
  outline: none; }

.cityOption:hover {
  cursor: pointer; }

/*# sourceMappingURL=CitySelect.stylesheet.css.map */
