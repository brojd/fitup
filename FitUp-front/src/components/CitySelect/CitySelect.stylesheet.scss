@import "../../../sass_variables/colors";
@import "../../../sass_variables/widths";

.citySelect {
  height: 17px;
  border: none;
  border-bottom: 2px solid rgba($guardsman_red, 0.2);
  color: #f9f9f9;
  background: linear-gradient(to bottom right, #400000, rgba(169, 0, 0, 0.85));
  border-radius: 3px;
  font-size: 11px;
  z-index: 5;
  width: 17%;
}

.citySelect:hover {
  cursor: pointer;
}

.citySelect:focus {
  outline: 0;
}

.cityOption {
  color: $black;
  background: $white;
  outline: none;
}

.cityOption:hover {
  cursor: pointer;
}
