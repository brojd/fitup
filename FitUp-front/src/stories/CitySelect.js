import React from 'react';
import { storiesOf} from '@kadira/storybook';
import CitySelect from '../components/CitySelect/CitySelect.component';
import GYMS from './mock_data/gyms';

storiesOf('CitySelect', module)
  .add('search bar for type of classes', () => (
    <CitySelect gyms={GYMS}/>
  ));

