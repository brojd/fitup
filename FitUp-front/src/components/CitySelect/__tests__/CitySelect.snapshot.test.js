import React from 'react';
import CitySelect from '../CitySelect.component';
import renderer from 'react-test-renderer';

test('CitySelect renders correctly with given props', () => {
  
  const gyms = [
    {
      'name': 'Fitness Academy',
      'address': {
        'city': 'Wrocław',
        'postal_code': '50-001',
        'street':'Długa',
        'number':3
      }
    },
    {
      'name': 'Fitness World',
      'address': {
        'city': 'Wrocław',
        'postal_code': '50-001',
        'street':'Olszowa',
        'number':3
      },
    },
    {
      'name': 'Spartan',
      'address': {
        'city': 'Opole',
        'postal_code': '50-001',
        'street':'Kwiska',
        'number':3
      }
    }
  ];
  const component = renderer.create(
    <CitySelect onChange={jest.fn()}
                gyms={gyms}
                userLocation='Gdańsk' />
  );
  let tree = component.toJSON();
  
  expect(tree).toMatchSnapshot();
});
