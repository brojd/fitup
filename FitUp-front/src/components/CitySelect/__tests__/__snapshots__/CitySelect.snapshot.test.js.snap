exports[`test CitySelect renders correctly with given props 1`] = `
<select
  className="uk-float-right citySelect"
  defaultValue="Gdańsk"
  onChange={[Function]}>
  <option
    className="cityOption"
    value="Wrocław">
    Wrocław
  </option>
  <option
    className="cityOption"
    value="Opole">
    Opole
  </option>
  <option
    className="cityOption"
    value="Gdańsk">
    Gdańsk
  </option>
</select>
`;
